### PR TITLE
rehearse: unify how changed jobs are detected

### PR DESCRIPTION
**Pull request by petr-muller**
The comparison logic was duplicated in methods that compared Presubmits
and Periodics, so I extracted the logic to a common method. I also
enhanced the logging to capture the reasons why a job is selected for
rehearsal.

### CODE REVIEW

File: pkg/diffs/diffs.go [hunk 1]
```diff
@@ -97,27 +98,24 @@ func GetChangedPresubmits(prowMasterConfig, prowPRConfig *prowconfig.Config, log
...
-				} else if masterJob.Optional && !job.Optional {
-					logger.WithFields(logFields).Info(chosenJob)
-					ret.Add(repo, job)
-				} else if !masterJob.AlwaysRun && job.AlwaysRun {
-					logger.WithFields(logFields).Info(chosenJob)
-					ret.Add(repo, job)
-				} else if masterJob.Cluster != job.Cluster {
-					logger.WithFields(logFields).Info(chosenJob)
-					ret.Add(repo, job)
+			var why []string
```
Comment (alvaroaleman) [inline pkg/diffs/diffs.go:101]: IMHO variables should describe what they contain, not what questions can be asked to which the content of the var is the correct answer, so something like `reasons` for the var name and log key makes IMHO more sense

Comment (alvaroaleman) [reply pkg/diffs/diffs.go:101]: (as in: relation var name var content should be an equasion)

Comment (petr-muller) [reply pkg/diffs/diffs.go:101]: :+1:

File: pkg/diffs/diffs.go [hunk 2]
```diff
@@ -23,6 +23,7 @@ import (
 const (
 	logRepo       = "repo"
 	logJobName    = "job-name"
+	logReasons    = "why"
```
Comment (alvaroaleman) [inline pkg/diffs/diffs.go:26]: here as well

**Comment by alvaroaleman** [comment]
Just to test it out:
/test ?

**Comment by petr-muller** [comment]
That's goddamn sweet.

File: pkg/diffs/diffs.go (hunk 2 above)
Comment (petr-muller) [reply pkg/diffs/diffs.go]: aye

**Pull request merged**